vendor/k8s.io/kubernetes/test/e2e/apimachinery/aggregator.go

```diff
@@ -115,6 +118,33 @@ func cleanTest(client clientset.Interface, aggrclient *aggregatorclient.Clientse
 	_ = client.RbacV1().ClusterRoleBindings().Delete(context.TODO(), "wardler:"+namespace+":sample-apiserver-reader", metav1.DeleteOptions{})
 }
 
+func ensureSampleAPIUnregisteredFromAllRootAPIs(client clientset.Interface, namespace string) {
+	apiHelper, err := NewAPIEndpointHelper(client, namespace)
+	if err != nil && err == UnsupportedPlatformErr {
+		framework.Logf("This test runs on a platform that we don't know how to detect all instances of Kube API server." + " " +
+			"That means it might be flaky as there is no deterministic way of ensuring the sample server was unregistered from all replicas."  + " " +
+			"You can ignore this message if you are running on a platform with single api server.")
+		return
+	}
+	if err != nil {
+		framework.ExpectNoError(err, "unable to create an API endpoint helper")
+	}
+	defer apiHelper.CleanUp()
+
+	apiHelper.WaitForAll(func(client *clientset.Clientset)(bool, error) {
```

Comment:
Maybe add an optional wait time for if the apiserver doesn't support checking 

```
apiHelper.WaitForAll(2*time.Minute, func() (bool, error) { ... })
```

And then you could only print the warning log when you call WaitForAll() and we have no endpoints, removing the need for this code to know about UnsupportedPlatformErr